auth.go

```diff
@@ -154,6 +160,13 @@ func WithJWTParserScopeFilter(scopeFilter jwt.ScopeFilter) JWTParserOption {
 	}
 }
 
+// WithHTTPUserAgent sets the User-Agent header for HTTP requests to JWKS endpoint.
+func WithHTTPUserAgent(userAgent string) JWTParserOption {
```

Comment:
Could you please rename `WithHTTPUserAgent` to `WithJWTParserHTTPUserAgent` for consistency with other JWTParser's options?